lib/runtime/wasmer/instance_test.go

```diff
@@ -63,3 +65,39 @@ func TestInstance_CheckRuntimeVersion(t *testing.T) {
 	require.Equal(t, expected.ImplVersion(), version.ImplVersion())
 	require.Equal(t, expected.TransactionVersion(), version.TransactionVersion())
 }
+
+func TestDecompressWasm(t *testing.T) {
+	encoder, _ := zstd.NewWriter(nil)
+	cases := []struct {
+		in       []byte
+		expected []byte
+		msg      string
+	}{
+		{
+			[]byte{82, 188, 83, 118, 70, 219, 142},
+			[]byte{82, 188, 83, 118, 70, 219, 142},
+			"partial compression flag",
+		},
+		{
+			[]byte{82, 188, 83, 118, 70, 219, 142, 6},
+			[]byte{82, 188, 83, 118, 70, 219, 142, 6},
+			"wrong compression flag",
+		},
+		{
+			[]byte{82, 188, 83, 118, 70, 219, 142, 6, 221},
+			[]byte{82, 188, 83, 118, 70, 219, 142, 6, 221},
+			"wrong compression flag with data",
+		},
+		{
+			append([]byte{82, 188, 83, 118, 70, 219, 142, 5}, encoder.EncodeAll([]byte("compressed"), make([]byte, 0))...),
```

Comment:
nit I think this should work?

```suggestion
			append([]byte{82, 188, 83, 118, 70, 219, 142, 5}, encoder.EncodeAll([]byte("compressed"), nil)...),
```